apps/etterlatte-vilkaarsvurdering/src/main/kotlin/no/nav/etterlatte/vilkaarsvurdering/VilkaarsvurderingService.kt

```diff
@@ -31,27 +33,40 @@ class VilkaarsvurderingService(
     private val vilkaarsvurderingRepository: VilkaarsvurderingRepository,
     private val behandlingKlient: BehandlingKlient,
     private val grunnlagKlient: GrunnlagKlient,
+    private val featureToggleService: FeatureToggleService,
 ) {
     private val logger = LoggerFactory.getLogger(VilkaarsvurderingService::class.java)
 
-    fun hentVilkaarsvurdering(behandlingId: UUID): Vilkaarsvurdering? = vilkaarsvurderingRepository.hent(behandlingId)
+    fun hentVilkaarsvurdering(behandlingId: UUID): Vilkaarsvurdering? {
+        return vilkaarsvurderingRepository.hent(behandlingId)
+    }
+
+    suspend fun hentBehandlingensGrunnlag(
+        behandlingId: UUID,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Grunnlag {
+        return hentDataForVilkaarsvurdering(behandlingId, brukerTokenInfo).second
+    }
 
     suspend fun oppdaterTotalVurdering(
         behandlingId: UUID,
         brukerTokenInfo: BrukerTokenInfo,
         resultat: VilkaarsvurderingResultat,
     ): Vilkaarsvurdering =
         tilstandssjekkFoerKjoering(behandlingId, brukerTokenInfo) {
+            val (behandling, grunnlag) = hentDataForVilkaarsvurdering(behandlingId, brukerTokenInfo)
             val virkningstidspunkt =
-                behandlingKlient.hentBehandling(behandlingId, brukerTokenInfo).let {
-                    it.virkningstidspunkt?.dato?.atDay(1)
-                } ?: throw IllegalStateException("Virkningstidspunkt må være satt for å sette en vurdering")
+                behandling.virkningstidspunkt?.dato?.atDay(1)
+                    ?: throw IllegalStateException("Virkningstidspunkt må være satt for å sette en vurdering")
```

Comment:
Kunne vært nice å ha denne som en `UgyldigForespoerselException`, slik at feilmeldingen i detail her blir med ut i feil-responsen

Reply:
Out of scope :)